platform/ios/src/MGLScaleBar.mm

```diff
@@ -235,7 +235,13 @@ - (void)updateVisibility {
     ? MGLMetricTable[count-1].distance
     : MGLImperialTable[count-1].distance;
     
-    self.alpha = maximumDistance > allowedDistance ? 0 : 1;
+    CGFloat alpha = maximumDistance > allowedDistance ? .0f : 1.0f;
+    
+    if(self.alpha != alpha) {
+        [UIView animateWithDuration:.2f delay:0 options:UIViewAnimationOptionBeginFromCurrentState animations:^{
```

Comment:
FYI, most of MGLMapView’s animations occur within 300 milliseconds, via `MGLAnimationDuration`. (200 milliseconds is fine, though.)